hashintel/hash
# Prepare design system for publishing

## 🌟 What is the purpose of this PR?

This PR improves the compatibility of `@local/design-system` with it being published on npm. Because we plan to add a build step, we should avoid imports from `@local/design-system/src/*` because a built version will need it to be `@local/design-system/dist/*`. We could switch over to live incremental rebuilds, it it might be best to tweak `package.json` before publishing for a slightly better DX. More PRs to follow, including https://github.com/hashintel/hash/pull/1953.

I renamed `src/ui.ts` to `src/main.ts` given the opportunity.




## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203543021352041/1203443872976464/f) _(internal)_

## ❓ How to test this?

1.  Try importing from `"@local/design-system/src/whatever`
1.  Observe ESLint error saying `Please import from @local/design-system instead`
